nixos/modules/services/networking/inadyn.nix

```diff
@@ -0,0 +1,205 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.inadyn;
+
+  # check if a value of an attrset is not null or an empty collection
+  nonEmptyValue = _: v: ! isNull v && v != [ ] && v != { };
+
+  renderOption = k: v:
+    if builtins.elem k [ "provider" "custom" ] then
+      lib.concatStringsSep "\n"
+        (mapAttrsToList
+          (name: config: ''
+            ${k} ${name} {
+                ${lib.concatStringsSep "\n    " (mapAttrsToList renderOption (filterAttrs nonEmptyValue config))}
+            }'')
+          v)
+    else if k == "include" then
+      "${k}(\"${v}\")"
+    else if k == "hostname" && builtins.isList v then
+      "${k} = { ${builtins.concatStringsSep ", " (map (s: "\"${s}\"") v)} }"
+    else if builtins.isBool v then
+      "${k} = ${boolToString v}"
+    else if builtins.isString v then
+      "${k} = \"${v}\""
+    else
+      "${k} = ${toString v}";
+
+  configFile' = pkgs.writeText "inadyn.conf"
+    ''
+    # This file was generated by nix
+    # do not edit
+
+    ${(lib.concatStringsSep "\n" (mapAttrsToList renderOption (filterAttrs nonEmptyValue cfg.settings)))}
+    '';
+
+  configFile = if (cfg.configFile != null) then cfg.configFile else configFile';
+in
+{
+  options.services.inadyn = with types;
+    let providerOptions =
+      {
+        include = mkOption {
+          default = null;
+          description = mdDoc "File to include additional settings for this provider from.";
+          type = nullOr path;
+        };
+        ssl = mkOption {
+          default = true;
+          description = mdDoc "Whether to use HTTPS for this DDNS provider.";
+          type = bool;
+        };
+        username = mkOption {
+          default = null;
+          description = mdDoc "Username for this DDNS provider.";
+          type = nullOr str;
+        };
+        password = mkOption {
+          default = null;
+          description = mdDoc ''
+            Password for this DDNS provider.
+
+            WARNING: This will be world-readable in the nix store.
+            To store credentials securely, use the `include` or `configFile` options.
+          '';
+          type = nullOr str;
+        };
+        hostname = mkOption {
+          default = "*";
+          example = "your.cool-domain.com";
+          description = mdDoc "Hostname alias(es).";
+          type = either str (listOf str);
+        };
+      };
+    in
+    {
+      enable = mkEnableOption (mdDoc ''
+        synchronise your machine's IP address with a dynamic DNS provider using inadyn
+      '');
+      interval = mkOption {
+        default = "*-*-* *:*:00";
```

Comment:
not sure if we want to run this every minute. ddclient is every 10 minues: https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/services/networking/ddclient.nix#L107

Reply:
This is the upstream default (when running as a daemon, at least) https://man.archlinux.org/man/inadyn.conf.5